src/Core/Services/CryptoService.cs

```diff
@@ -152,9 +152,13 @@ public Task ClearMasterKeyAsync(string userId = null)
             return _stateService.SetMasterKeyAsync(null, userId);
         }
 
-        public async Task<Tuple<UserKey, EncString>> EncryptUserKeyWithMasterKeyAsync(MasterKey masterKey)
+        public async Task<Tuple<UserKey, EncString>> EncryptUserKeyWithMasterKeyAsync(MasterKey masterKey, UserKey userKey = null)
         {
-            var userKey = await GetUserKeyAsync() ?? await MakeUserKeyAsync();
+            userKey ??= await GetUserKeyAsync();
+            if (userKey == null)
+            {
+                throw new Exception("No user key found");
```

Comment:
⛏️ If we need to spot this in AppCenter to add the appropriate fix I'd go for a custom exception like `UserAndMasterKeysNullException` was used in another PR so it can be easily spotted on AppCenter. We could have one exception `KeysNullException(types of key)` so it changes the message depending on the keys missing. Thoughts?

Reply:
After talking with @jlf0dev we decided to add classes for each null key to be easier to track by reference in code.